contracts/liquidityMining/KyberFairLaunch.sol

```diff
@@ -389,73 +376,64 @@ contract KyberFairLaunch is IKyberFairLaunch, PermissionAdmin, ReentrancyGuard {
     pool.lastRewardBlock = lastAccountedBlock;
   }
 
-  /**
-  * @dev check if a pool of _stakeToken exists and not ended yet, only called when adding a new pool
-  * @param _stakeToken: token to check for exisiting pool
-  */
-  function isDuplicatedPool(address _stakeToken) public view returns (bool) {
-    uint256 _pid = latestTokenPoolId[_stakeToken];
-    // not exist
-    if (_pid == 0) return false;
-    _pid -= 1;
-    // if already ended -> consider as not duplicated
-    // if the pool has not been started, use updatePool function to update pool info
-    return (poolInfo[_pid].endBlock >= block.number);
-  }
-
   /**
   * @dev withdraw _amount of stakeToken from pool _pid, also harvest reward for the sender
   */
   function _withdraw(uint256 _pid, uint256 _amount) internal {
-    require(_pid < poolLength, 'withdraw: invalid pool id');
 
     PoolInfo storage pool = poolInfo[_pid];
     UserInfo storage user = userInfo[_pid][msg.sender];
     require(user.amount >= _amount, 'withdraw: insufficient amount');
 
     // update pool reward and harvest
     updatePoolRewards(_pid);
-    _harvest(msg.sender, _pid);
+    _updateUserReward(msg.sender, _pid, true);
 
     user.amount = user.amount.sub(_amount);
     pool.totalStake = pool.totalStake.sub(_amount);
 
-    if (pool.stakeToken != address(0)) {
-      IERC20Ext(pool.stakeToken).safeTransfer(msg.sender, _amount);
-    }
+    IERC20Ext(pool.stakeToken).safeTransfer(msg.sender, _amount);
 
     emit Withdraw(msg.sender, _pid, block.number, user.amount);
   }
 
   /**
-  * @dev harvest reward of _to address from pool _pid
+  * @dev update reward of _to address from pool _pid, harvest if needed
   */
-  function _harvest(address _to, uint256 _pid) internal {
+  function _updateUserReward(address _to, uint256 _pid, bool shouldHarvest) internal {
     PoolInfo storage pool = poolInfo[_pid];
```

Comment:
better if we only have variable like `lastRewardPerShare`
```solidity
uint256 lastRewardPerShare = poolInfo[_pid].accRewardPerShare
```

Reply:
can use `accRewardPerShare` to differentiate between user and pool.